ci-operator/step-registry/cucushift/hypershift-extended/nodepool-image/azure/get-marketplace-image/cucushift-hypershift-extended-nodepool-image-azure-get-marketplace-image-commands.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+if [[ -z $HYPERSHIFT_AZURE_MARKETPLACE_IMAGE_PUBLISHER ]]; then
+    echo "\$HYPERSHIFT_AZURE_MARKETPLACE_IMAGE_PUBLISHER is empty, skip"
+    exit 0
+fi
+
+AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/osServicePrincipal.json"
+AZURE_AUTH_CLIENT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientId)"
+AZURE_AUTH_CLIENT_SECRET="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientSecret)"
+AZURE_AUTH_TENANT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .tenantId)"
+AZURE_LOCATION="$LEASED_RESOURCE"
+
+az --version
+az login --service-principal -u "${AZURE_AUTH_CLIENT_ID}" -p "${AZURE_AUTH_CLIENT_SECRET}" --tenant "${AZURE_AUTH_TENANT_ID}" --output none
+
+set -x
+
+IMAGE_JSON=/tmp/azure-marketplace-image.json
+IMAGE_SKU_FILE="${SHARED_DIR}"/azure-marketplace-image-sku
+IMAGE_VERSION_FILE="${SHARED_DIR}"/azure-marketplace-image-version
+PULL_SECRET="${CLUSTER_PROFILE_DIR}"/pull-secret
+WRITABLE_PULL_SECRET=/tmp/pull-secret
+
+echo "Getting pull secret"
+cp "$PULL_SECRET" "$WRITABLE_PULL_SECRET"
+KUBECONFIG="" oc registry login --to "$WRITABLE_PULL_SECRET"
+
+echo "Extracting OCP version from release image"
+OCP_MAJOR_VERSION=$(oc adm release info "$RELEASE_IMAGE_LATEST" -a "$WRITABLE_PULL_SECRET" -o json | jq -r '.metadata.version' | cut -d . -f 1)
+OCP_MINOR_VERSION=$(oc adm release info "$RELEASE_IMAGE_LATEST" -a "$WRITABLE_PULL_SECRET" -o json | jq -r '.metadata.version' | cut -d . -f 2)
+
+echo "Extracting Azure marketplace image info"
+az vm image list \
+    --architecture "$HYPERSHIFT_AZURE_MARKETPLACE_IMAGE_ARCH" \
+    --publisher "$HYPERSHIFT_AZURE_MARKETPLACE_IMAGE_PUBLISHER" \
+    --offer "$HYPERSHIFT_AZURE_MARKETPLACE_IMAGE_OFFER" \
+    --sku "aro_${OCP_MAJOR_VERSION}${OCP_MINOR_VERSION}" \
```

Comment:
I don't know that this is standardized. I just look and saw several variations `415-v2, 417-arm, aro_415` etc.

Reply:
I'd like to dynamically select the SKU based on the HC's version and use the `aro_XY` naming for now.

If the naming convention changes in the future, we can introduce an environment variable to control the naming scheme, allowing for options like `aro_XY`, `XY-arm`, etc.

Reply:
Ok that's fine. Just wanted to make sure you were aware there were other variations.

Reply:
Yup, thanks for the info